discovery-provider/src/tasks/entity_manager/entities/user.py

```diff
@@ -333,12 +328,45 @@ def update_user_metadata(
             challenge_event_bus,
         )
 
-    if metadata.get("allow_ai_attribution"):
-        user_record.allow_ai_attribution = metadata["allow_ai_attribution"]
-
     return user_record
 
 
+# get previous CIDData and merge new metadata into it
+# this is to support collectibles, associated_wallets which aren't being indexed yet
+# once those are indexed and backfilled this can be removed
+def merge_user_metadata(params: ManageEntityParameters, user_record: User):
+    prev_cid_response = (
+        params.session.query(CIDData)
+        .filter_by(
+            cid=user_record.metadata_multihash,
+        )
+        .first()
+    )
+
+    if prev_cid_response:
+        # merge previous and current metadata
+        updated_metadata = prev_cid_response.data | params.metadata
+
+        # generate a cid
+        updated_metadata_cid = str(
+            generate_metadata_cid_v1(json.dumps(updated_metadata))
+        )
+
+        # save the metadata blob
+        cid_data = CIDData(
+            cid=updated_metadata_cid,
+            type="user",
+            data=updated_metadata,
+        )
+        params.session.merge(cid_data)
```

Comment:
is this session.merge duplicated? i think we're already doing this here

```
            save_cid_metadata(session, cid_metadata, cid_type)
```

and as a side note not your change but i think we should just be adding cid_data to records_to_save and bulk writing at once instead of separately doing session.merge. think it's better perf